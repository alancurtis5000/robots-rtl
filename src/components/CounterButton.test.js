import { render, cleanup, fireEvent  } from '@testing-library/react';
import CounterButton from './CounterButton';

describe("CounterButton",()=>{
    afterEach(cleanup);

    it('Should render',()=>{
        const mockColor = 'red'
        const {asFragment} = render(<CounterButton color={mockColor} />);
        expect(asFragment()).toMatchSnapshot();
    })
    it('correctly increments the counter', ()=>{
        const mockColor = 'red';
        const {getByTestId} = render(<CounterButton color={mockColor} />);
        const button = getByTestId('counter');
        fireEvent.click(button)
        const result = button.textContent
        expect(result).toBe('Count: 2')
    })
})