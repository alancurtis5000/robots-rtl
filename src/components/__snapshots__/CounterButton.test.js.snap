// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Should render 1`] = `
<DocumentFragment>
  <button
    color="red"
    data-testid="counter"
  >
    Count: 1
  </button>
</DocumentFragment>
`;
